engine/modules/camera/src/main/java/com/codingame/gameengine/module/camera/CameraModule.java

```diff
@@ -0,0 +1,163 @@
+ */
+@Singleton
+public class CameraModule implements Module {
+
+    GameManager<AbstractPlayer> gameManager;
+    @Inject
+    GraphicEntityModule entityModule;
+    Map<Integer, Boolean> registered, newRegistration;
+    ArrayList<Entity<?>> trackedEntities;
+    Double cameraOffset;
+    Integer container, sizeX, sizeY = -1;
+    boolean sentContainer = false;
+    double previousOffset = -1;
+    boolean active = true;
+    boolean oldActive = false;
+
+
+    @Inject
+    CameraModule(GameManager<AbstractPlayer> gameManager) {
+        trackedEntities = new ArrayList<>();
+        this.gameManager = gameManager;
+        gameManager.registerModule(this);
+        registered = new HashMap<>();
+        newRegistration = new HashMap<>();
+        cameraOffset = 10.;
+    }
+
+    @Override
+    public void onGameInit() {
+        sendFrameData();
+    }
+
+    @Override
+    public void onAfterGameTurn() {
+        sendFrameData();
+    }
+
+    @Override
+    public void onAfterOnEnd() {
+    }
+
+
+    private void sendFrameData() {
+        Object[] data = {null, null, null, null};
+        Object[] empty = {null, null, null, null};
+        if (!newRegistration.isEmpty()) {
+            data[0] = new HashMap<>(newRegistration);
+            newRegistration.clear();
+        }
+        if (cameraOffset != previousOffset) {
+            data[1] = cameraOffset;
+            previousOffset = cameraOffset;
+        }
+        if (!sentContainer && container >= 0) {
+            data[2] = new Integer[]{container, sizeX, sizeY};
+            sentContainer = true;
+        }
+        if (oldActive != active) {
+            oldActive = active;
+            data[3] = active;
+        }
+        if (!Arrays.equals(data, empty)) {
+            gameManager.setViewData("c", data);
+        }
+    }
+
+    private boolean isContainerChild(Entity<?> entity, int _container) {
+        boolean t = false;
+        Entity<?> root = entity;
+        while ((!t) && root.getParent().isPresent()) {
+            root = root.getParent().get();
+            t = root.getId() == _container;
+        }
+        return t;
+    }
+
+    /**
+     * Make the camera include the entity in its field of view
+     *
+     * @param entity the <code>Entity</code> to add to the tracked entities
+     */
+    public void addTrackedEntity(Entity<?> entity) {
+        if (isContainerChild(entity, container)) {
+            int id = entity.getId();
+            trackedEntities.add(entity);
+            if (!registered.getOrDefault(id, false)) {
+                newRegistration.put(id, true);
+                registered.put(id, true);
+            }
+        } else {
+            throw new RuntimeException("The entity given can't be track because it's not the child of " +
+                    "the container / on of the container child !\n" +
+                    "Don't forget to init the camera with the setContainer method");
```

Comment:
Error message is a little off. Could be:

```java
            throw new RuntimeException("The entity given can't be tracked because it's not a child of the container!\n" +
                    "Don't forget to init the camera with the setContainer method");
```